### ISSUE TITLE
Regression introduced in 1.1.198 involving resolving T from Type[T]

### ISSUE THREAD

**Issue by rsokl** [microsoft/pyright#2748]
I found a regression in 1.1.198 that may be related to the change in behavior pertaining to allowing `type` to be assigned to `Type[T]`. It is demonstrated by the following code.

```python
from typing import TypeVar, Type, Union, Protocol

T = TypeVar("T", covariant=True)

T1 = TypeVar("T1")
T2 = TypeVar("T2")


class A(Protocol[T]): 
    ...


def f(x: T1) -> Type[A[T1]]:
    raise NotImplemented()

def g(x: Union[A[Type[T2]], Type[A[Type[T2]]]]) -> T2:
    raise NotImplemented()

reveal_type(g(f(int)))
# Previous behavior: x is revealed as "int"
# New behavior: x is revealed as "T1@g"
```

I should mention that in `hydra-zen` this regression manifests a bit differently.

```python
from hydra_zen import builds, instantiate

reveal_type(instantiate(builds(int)))
# Previous behavior: x is revealed as "int"
# New behavior: x is revealed as "Any"
```

I can't quite figure out why in the toy example the type is "T1@g", whereas in my applied case the type is "Any". If it useful here are links to the signatures: [builds](https://github.com/mit-ll-responsible-ai/hydra-zen/blob/189b20a944f6e84026b67e8744cc16f9ff025a2e/src/hydra_zen/structured_configs/_implementations.py#L686-L742), [instantiate](https://github.com/mit-ll-responsible-ai/hydra-zen/blob/189b20a944f6e84026b67e8744cc16f9ff025a2e/src/hydra_zen/_hydra_overloads.py#L44-L85). There are also some cases where the revealed type has become "Unknown" rather than "Any.

A side note: the union that I use in the signature of `g`  appears to bypass the regression that I reported in #2746 .

**Comment by erictraut** [comment]
The underlying cause of this is the same as #2746. It will be addressed in the next release.

**Comment by rsokl** [comment]
Ah, I apologies for double posting. I couldn't tell that this was the same underlying issue. Thank you!

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.199, which I just published. It will also be included in the next release of pylance.
